src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -305,9 +305,16 @@ private static string NormalizeGuidCollectionItems(string bracketLiteralText)
 
             for (int i = 0; i < items.Length; i++)
             {
-                if (items[i][0] != '\'' && items[i][0] != '"')
+                if (items[i] == "null")
                 {
-                    items[i] = String.Format(CultureInfo.InvariantCulture, "'{0}'", items[i]);
+                    items[i] = String.Format(CultureInfo.InvariantCulture, "{0}", items[i]);

```

Comment:
Do you need to call `String.Format` if `items[i]` has a value of `null`? 
The above statement evaluates to:
```csharp
items[i] = items[i] = String.Format(CultureInfo.InvariantCulture, "{0}", "null");
```
and the value of `items[i]` will still be `"null"` after its executed.

Reply:
this is a part of the change..what will be returned will be "[null]" as opposed to "['null']"

Reply:
@ElizabethOkerio In that case maybe the statement should be:
```csharp
items[i] = null;
```
?
All the more reason why you need some tests...

Reply:
Not a 100% sure though...

Reply:
@gathogojr  before this change, this method was transforming null GUIDS to: 
``` 
" ['null']"
```
Which led to this NULL GUIDS being read as "null" in subsequent operations: 

This change transforms a null GUID to:
``` 
" [null]"
```
which will be read as a null value in subsequent operations:

Reply:
Can you have null and something else? E.g. [null,'844f0fb2-580d-4842-b965-bedc341736b3'] for example?

Reply:
Yes..